cmd/settlement_gemini.go

```diff
@@ -0,0 +1,318 @@
+		"the completed mass pay transaction id")
+	must(viper.BindPFlag("txn-id", geminiSettlementCmd.PersistentFlags().Lookup("txn-id")))
+	must(viper.BindEnv("txn-id", "TXN_ID"))
+	must(transformGeminiSettlementCmd.MarkPersistentFlagRequired("txn-id"))
+
+	transformGeminiSettlementCmd.PersistentFlags().StringVarP(&oauthClientID, "gemini-client-id", "g", "",
+		"the oauth client id needed to check that the user authorized the payment")
+	must(viper.BindPFlag("gemini-client-id", geminiSettlementCmd.PersistentFlags().Lookup("gemini-client-id")))
+	must(viper.BindEnv("gemini-client-id", "GEMINI_CLIENT_ID"))
+	must(transformGeminiSettlementCmd.MarkPersistentFlagRequired("gemini-client-id"))
+}
+
+// func geminiValidateResponse(
+// 	transactions *[]gemini.PayoutPayload,
+// 	response *[]gemini.PayoutResult,
+// ) (map[string]gemini.PayoutPayload, error) {
+// 	if len(*transactions) != len(*response) {
+// 		return nil, errors.New("response count did not match request count")
+// 	}
+// 	mappedTransactions := convertTransactionListIntoMap(transactions)
+// 	return mappedTransactions, nil
+// }
+
+func geminiSiftThroughResponses(
+	originalTransactions map[string]settlement.Transaction,
+	response *[]gemini.PayoutResult,
+) map[string][]settlement.Transaction {
+	transactions := make(map[string][]settlement.Transaction)
+
+	for _, payout := range *response {
+		if payout.Result == "Error" {
+			transactions["failed"] = append(transactions["failed"], originalTransactions[payout.TxRef])
+		} else {
+			if *payout.Status == "Pending" {
+				transactions["pending"] = append(transactions["pending"], originalTransactions[payout.TxRef])
+			} else {
+				transactions["completed"] = append(transactions["completed"], originalTransactions[payout.TxRef])
+			}
+		}
+	}
+	return transactions
+}
+
+// func convertTransactionListIntoMap(
+// 	transactions *[]gemini.PayoutPayload,
+// ) map[string]gemini.PayoutPayload {
+// 	transactionMap := make(map[string]gemini.PayoutPayload)
+// 	for _, payoutRequest := range *transactions {
+// 		transactionMap[payoutRequest.TxRef] = payoutRequest
+// 	}
+// 	return transactionMap
+// }
+
+// GeminiUploadSettlement marks the settlement file as complete
+func GeminiUploadSettlement(inPath string, signatureSwitch int, allTransactionsFile string, outPath string) error {
+	if outPath == "./gemini-settlement" {
+		// use a file with extension if none is passed
+		outPath = "./gemini-settlement-complete.json"
+	}
+
+	ctx := context.Background()
+	bulkPayoutFiles := strings.Split(inPath, ",")
+	geminiClient, err := gemini.New()
+	if err != nil {
+		return err
+	}
+
+	// transactionsForEyeshade := []settlement.Transaction{}
+	// failedTransactions := []settlement.Transaction{}
+	if allTransactionsFile == "" {
+		return errors.New("unable to upload without a transactions file to check against")
+	}
+	submittedTransactions := make(map[string][]settlement.Transaction)
```

Comment:
How big will these files be?  Are we worried about reading the full list of transactions in and processing on all of these all at once?

Reply:
if we are, i think we can just reduce the inputs and do them more frequently. some can be a couple dozen megabytes

Reply:
:+1: